src/trial_synth/who/transform.py

```diff
@@ -0,0 +1,104 @@
+import logging
+
+from lxml import etree
+import pandas as pd
+from tqdm import tqdm
+from pathlib import Path
+
+import bioregistry
+
+from .config import FIELDS, CSV_COLUMN_PATH
+from .trial_model import WHOTrial
+from .util import PREFIXES, makelist, transform_mappings, make_str
+from .validate import is_valid
+
+logger = logging.getLogger(__name__)
+
+
+def transform_csv_data(path: Path) -> pd.DataFrame:
+    """Transforms WHO data from CSV into a DataFrame
+
+    Parameters
+    ----------
+    path : Path
+        CSV data file path
+    
+    Returns
+    -------
+    pd.DataFrame
+        Transformed WHO data
+    """
+
+    # manual progress bar as pandas does not support tqdm natively
```

Comment:
This could be reimplemented such that you simply use the `csv` module to read the file line by line and in that same loop you build up your `rows = []` list.